### PR TITLE
The Hungry Geek - New blog

### PR DESCRIPTION
**Pull request by jodiedoubleday**
### What does this PR do?
* Creates a new Hungry Geek Blog and screens for public GitHub repos
* Allows users of all background to write, review and publish blog posts
* New design to our blog posts including all Meta data and Schema for better SEO ranking
* Moves a selection of old blog posts onto the new platform (others to follow if required)

### What unit or integration tests does this PR have?
None

### What selenium tests does this PR have?
None

### Where should the reviewer start?
* See the README file for how to set up the blog locally
* Try writing a draft blog post or reviewing live blog posts.

### Any background context you want to provide?
[The Hungry Geek](http://hungrygeek.holidayextras.co.uk/) blog is poorly designed, poorly managed, under used and doesn't fit in with the way HX works (none of us use wordpress).
I've been working n this project in my spare time in the hope of revitalising people interest in writing blog posts.

### What are the relevant tickets?
None

### What gif best describes how you feel about this work?
it's alive

### Quality checklist (for PR author to complete BEFORE code review):

- [x] I've checked there is appropriate unit test coverage.
- [x] I've checked there is appropriate selenium test coverage.
- [x] I've included regression tests for any bug fixes.
- [x] I've updated documentation with any changes to procedures.
- [x] I've tested this on mobile Android, Safari, Chrome for any visual changes.
- [x] I've tested this on desktop IE9/10/11, Firefox, Chrome for any visual changes.
- [x] I've checked this work against the requirements of the Jira.
- [x] I've informed those who need to know of any unforeseen impact of this work (database migrations, deployment procedure changes).
- [x] I am ready for this to be code reviewed, merged and tested.

### IP or Developer Review:

- [x] I agree with the assumptions made in the quality checklist.
- [x] I’ve witnessed the work behaving as expected.
- [x] I’ve run all the tests locally and they pass.
- [x] +1 from me!

### Software Engineer or Developer review:

- [x] I agree with the assumptions made in the quality checklist.
- [x] I’ve witnessed the work behaving as expected.
- [x] I’ve run all the tests locally and they pass.
- [x] I’ve checked for appropriate test coverage.
- [x] I’ve checked for coding anti-patterns.
- [x] I've checked this work against the requirements of the Jira.
- [x] +1 from me!

### Software Engineer or project guru review:

- [x] I agree with the assumptions made in the quality checklist.
- [x] I’ve witnessed the work behaving as expected.
- [x] I’ve run all the tests locally and they pass.
- [x] I’ve checked for appropriate test coverage.
- [x] I’ve checked for coding anti-patterns.
- [x] I don’t believe this work introduces unnecessary technical debt.
- [x] +1 from me!

### CODE REVIEW

**Comment by dancork** [comment]
Taking dev review

**Comment by hx-markterry** [comment]
why are files named .html when they are css or in a templating language?

File: _includes/head.html [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width initial-scale=1">
```
Comment (dancork) [inline _includes/head.html]: Should this not be comma separated? 

'<meta name="viewport" content="width=device-width, initial-scale=1">'

File: _includes/styles.html [hunk 2]
```diff
@@ -0,0 +1,284 @@
+<!-- For critical styles -->
+<!-- No PX Values -->
+<!-- RGBA Colors only -->
+
+<link href='http://fonts.googleapis.com/css?family=Roboto:400,300,100,500,700|Yanone+Kaffeesatz:400,700,300,200' rel='stylesheet' type='text/css'>
```
Comment (dancork) [inline _includes/styles.html]: Seems a lot of fonts to be including. For page speed might be better to limit the number of font-weights.

File: _includes/styles.html [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+
+<style>
```
Comment (dancork) [inline _includes/styles.html]: Any reason this not in a css file?

**Comment by jodiedoubleday** [comment]
@hx-markterry Jekyll compiled `.html` and `.md` files into static sites. There could be an argument to convert some of these `.html` files into `.md` files. but I'm not sure its necessary.

**Comment by jodiedoubleday** [comment]
Hello @danjenkins 
I haven't consciously removed people not employed by HX, I believe there are many missing posts (some from Alanna and myself for example). I will be adding those in as soon as I get chance.

As for redirects, Jekyll uses the same url structure with dates - 
/2014/06/05/avoiding-common-pitfalls-starting-node-js/ for example.
but i will ensure this follows through to maintain SEO rankings.

**Comment by dancork** [comment]
You can customise the permalink style, if needs be.

http://jekyllrb.com/docs/permalinks/#template-variables

File: _includes/styles.html [hunk 4]
```diff
@@ -0,0 +1,284 @@
+<!-- For critical styles -->
+<!-- No PX Values -->
+<!-- RGBA Colors only -->
+
+<link href='http://fonts.googleapis.com/css?family=Roboto:300,700|Yanone+Kaffeesatz:200,400,700' rel='stylesheet' type='text/css'>
+
+<style>
```
Comment (dancork) [inline _includes/styles.html]: Is there any reason that this isn't in it's own .css file?

**Comment by Joezo** [comment]
I'm not entirely convinced by prose.io as advised in the readme. It concerns me that it has access to all public/private repos. There is a considerable number of open issues on it's repo and that doesn't appear to have been updated in months. While we could change the readme to link to the authorize link without requesting private repo access it still seems a risk.

I understand the need for something like this though, it enables people who are unfamiliar with markdown  and github to write posts easier. To get around the lack of markdown knowledge, there a few editors such as [MacDown](http://macdown.uranusjr.com/), that are available to download for free, which is great. 

What do others think?

**Comment by hx-markterry** [comment]
for the sake of a pretty editor, there are many alternatives that won't need to be linked to someone's github account. the markdown hurdle is such as small one i dont think using an external service here is needed.

**Comment by jodiedoubleday** [comment]
Addressing @Joezo's issue with Prose.io. I have updated the readme with a link that only allows prose to access public repos. This should address some of the concerns.

**Comment by jodiedoubleday** [comment]
@dancork Styles isn't in its own CSS file, because I wanted to be highly rated on [google page speed insights](https://developers.google.com/speed/pagespeed/insights/).
Critical CSS is inline in the `<head>` of the doc, as the styles of this blog are relatively small, I decided all css is critical. That and I couldn't get [critical](https://github.com/addyosmani/critical) to work properly.

**Comment by dancork** [comment]
Makes sense, guess it's something that can be looked in to in the future. Better to get the thing up and running.

**Comment by hxpaul** [comment]
Agree with Mark T and Joe I think, don't think we need prose.io here. You don't need to know markdown to be able to write a post, only to do advanced styling of it really. If you can hit the return key you can format your post enough to make it readable. Maybe we should say "if enough people can't post, then let's find a wysiwyg editor" later.

**Comment by jodiedoubleday** [comment]
I don't want to restrict this to Dev's only, I have had interest from product owners and scrum masters who want to be able to write a blog post on our website without having to learn how to create a branch, commit, put in a PR etc.
prose.io (or any other wsywyg editor, I'm not fussy) solves this issue for them. Everyone else will continue to create new branches, PR etc.

**Comment by Joezo** [comment]
I actually think it's a good opportunity for the POs and scrum masters to learn how to do those things. I imagine it would be useful for them to have a small understanding of how git works. It wouldn't take long for one of us to help out, there are plenty of us :)

**Comment by dancork** [comment]
We could do a session with them explaining the basics of git. I feel a little dirty suggesting it but if they're daunted by the command line they can always use the Github GUI.

**Comment by jodiedoubleday** [comment]
@Joezo I agree it's useful, I would love everyone to learn how to do it, but in reality it just wont happen.
If you want to start by showing Ant how to pull a branch, commit and put in a PR go for it!

@dancork Github GUI or [StackedIt.io](https://stackedit.io) also requires access to all our repos, only difference is its run by GitHub themselves, but its designed for coding rather than blog writing.

**Comment by jodiedoubleday** [comment]
How about this compromise...
For those who don't know how to pull, commit and PR a branch with Git, will be restricted to ONLY the holidayextras.github.io repo, therefore if the give prose.io permission to GitHub, prose will only see that repo. (I'll create a new "bloggers" group on GitHub)
This solves the issue of security around our private repos.

I update the README to push people to write blogs in the technical way if they know how. if they don't then we will restrict their GitHub access to the holidayextras.github.io, and if they need access to any other repo then they need to learn how to use git.

**Comment by Joezo** [comment]
Sounds good :)

**Comment by cameronviner** [comment]
Taking the IP/Dev review.

File: README.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+2. Github
+
+Just like any normal dev work, you can review a PR, comment on the content or editorial parts of the blog post and publish when we have 2 x :+1:'s
+
+--
+
+## Technical
+### Jekyll
+Jekyll is a static site generator - No more databases, comment moderation, or pesky updates to install — just your content.
+It uses HTML files for sturcture and Markdown files for content.
```
Comment (hx-markterry) [inline README.md]: spell casting: s.t.r.u.c.t.u.r.e

(remember the old kids tv game show "knightmare"?)

File: README.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+#### To Install
+```ruby
+gem install jekyll
+```
+
+#### To Run
+```
+jekyll serve
+```
+The Holiday Extras blog should now be running on your machine at http://127.0.0.1:4000/blog/
```
Comment (cameronviner) [inline README.md]: This address seems to give me a 404 but http://127.0.0.1:4000/ serves up the blog fine.

File: _includes/footer.html [hunk 7]
```diff
@@ -0,0 +1,6 @@
+<footer>
+  <div class="container">
+    <small>Published by <a href="http://hungrygeek.holidayextras.co.uk/" itemprop="publisher">Hungry Geek - Holiday Extras</a></small>
+    <small>Copyright &copy; 2014 <a href="http://www.holidayextras.co.uk/" itemprop="copyrightHolder">Holiday Extras</a>. All Rights Reserved.</small>
```
Comment (hx-markterry) [inline _includes/footer.html]: This is now out of date, can we make this dynamic as to the year of the current post?

**Comment by hx-markterry** [comment]
Good, only a few comments, will wait for responses before giving final ++

**Comment by cameronviner** [comment]
Looks good +1

**Comment by hx-markterry** [comment]
++

**Pull request merged**